src/jack.c

```diff
@@ -476,3 +503,80 @@ jalv_backend_activate_port(Jalv* jalv, uint32_t port_index)
 	}
 #endif
 }
+
+int
+jack_initialize (jack_client_t *client, const char *load_init)
+{
+	Jalv* const jalv = malloc(sizeof(Jalv));
+	if (jalv == NULL) {
+		fprintf(stderr, "Cannot allocate memory for JALV instance: %s\n", strerror(errno));
+		return -1;
+	}
+	memset(jalv, '\0', sizeof(Jalv));
+
+	/* backend is not yet allocated and has to be allocated */
+	jalv->backend = (JalvBackend*)calloc(1, sizeof(JalvBackend));
+	if (jalv->backend == NULL) {
+		return -1;
+	}
+	jalv->backend->client = client;
+	jalv->backend->is_internal_client = true;
+
+	printf("JALV args %s\n", load_init);
+
+	/* copy the init string becasue it will be manipulated by strtok() */
+	const size_t args_length = strlen(load_init) + 1;
+	char args[args_length];
+	strncpy(args, load_init, sizeof(args));
+	args[sizeof(args)-1] = '\0';
+
+	const char delimiters[] = " \t";
+	const int delimiter_count = sizeof(delimiters) - 1;
+
+	/* count the number of delimiters to find the worst case size of argv */
+	int delimiter_occurrence = 0;
+	for (int i=0; i<args_length; i++) {
+		for (int k=0; k<delimiter_count; k++) {
+			if (args[i] == delimiters[k]) {
+				delimiter_occurrence++;
+			}
+		}
+	}
+
+	/* Two additional arguments are required compared to the delimiter count
+	 * One is the first arguemnt which represense the process name and
+	 * the second one is required because a delimiter separates always two arguments
+	 */
+	const size_t argc_max = delimiter_occurrence + 2;
+	char* argv[argc_max];
+
+	int argc = 0;
+	/* add a dummy process name to the arguments */
+	argv[argc++] = "jack-jalv";
+
+	char* arg = strtok(args, delimiters);
+	while (arg != NULL) {
+		argv[argc++] = arg;
+		arg = strtok(NULL, delimiters);
+
+		assert(argc <= argc_max);
+	}
+
+
+	return jalv_open(jalv, argc, argv);
```

Comment:
I changed the behaviour. `jalv_open()` will call `jalv_close()`on a failure. This will free `jalv->backend`. `jalv` will be freed with an additional if-clause.